src/js/base/PersistForm.js

```diff
@@ -0,0 +1,34 @@
+import { useFormikContext } from "formik";
+import { debounce } from "lodash-es";
+import { useCallback, useLayoutEffect } from "react";
+import { useFirstRender } from "../utils/hooks";
+import { getSessionStorage, setSessionStorage, clearSessionStorageItem } from "../utils/utils";
+
+export const PersistForm = ({ formName, castValues }) => {
+    const formikContext = useFormikContext();
+    const firstRender = useFirstRender();
+    const debouncedSetSessionStorage = useCallback(debounce(setSessionStorage, 500), []);
+    useLayoutEffect(() => {
+        if (firstRender) {
+            const clearFormKey = getSessionStorage("clearForm");
+            if (clearFormKey) {
+                clearSessionStorageItem(clearFormKey);
+                clearSessionStorageItem("clearForm");
+            }
+
+            const prevFormData = getSessionStorage(formName);
+            if (prevFormData) {
+                const castFormData = castValues ? castValues(getSessionStorage(formName)) : getSessionStorage(formName);
+                formikContext.setValues(castFormData);
+            }
+        } else {
+            debouncedSetSessionStorage(formName, formikContext.values);
+        }
+    });
```

Comment:
You can possibly get rid of `useFirstRender` by splitting this into two hook calls.

Passing an empty array to `useEffect` as the second argument should cause the effect to be run only on first render.

https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects